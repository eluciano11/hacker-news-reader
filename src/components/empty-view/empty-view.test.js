import React from "react";
import { render, cleanup } from "@testing-library/react";

import EmptyView from "./empty-view";

afterEach(cleanup);

describe("Empty View Component Tests", () => {
  it("Should render empty view without content", () => {
    const { asFragment } = render(<EmptyView title="" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render empty view without links", () => {
    const { asFragment } = render(<EmptyView title="Empty view test" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render empty view with links", () => {
    const { asFragment } = render(
      <EmptyView
        title="Empty view test"
        subtitle="Subtitle test"
        url="https://hackernews.com"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
