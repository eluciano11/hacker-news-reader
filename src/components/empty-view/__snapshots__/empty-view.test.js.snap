// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty View Component Tests Should render empty view with links 1`] = `
<DocumentFragment>
  <div
    class="app__text--center"
  >
    <span
      aria-label="emoji"
      class="emoji"
      role="img"
    >
      ðŸ˜”
    </span>
    <h3
      class="title app__text--center"
    >
      Empty view test
    </h3>
    <a
      class="app__link"
      href="https://hackernews.com"
    >
      Subtitle test
    </a>
  </div>
</DocumentFragment>
`;

exports[`Empty View Component Tests Should render empty view without content 1`] = `
<DocumentFragment>
  <div
    class="app__text--center"
  >
    <span
      aria-label="emoji"
      class="emoji"
      role="img"
    >
      ðŸ˜”
    </span>
    <h3
      class="title app__text--center"
    />
  </div>
</DocumentFragment>
`;

exports[`Empty View Component Tests Should render empty view without links 1`] = `
<DocumentFragment>
  <div
    class="app__text--center"
  >
    <span
      aria-label="emoji"
      class="emoji"
      role="img"
    >
      ðŸ˜”
    </span>
    <h3
      class="title app__text--center"
    >
      Empty view test
    </h3>
  </div>
</DocumentFragment>
`;
